### PR TITLE
Refactor Uniswap v2 add/remove liquidity and add forge tests

### PR DESCRIPTION
**Pull request by Drblessing**
Add testing for add/removing liquidity as well as show how to mint ERC20 tokens in forge tests.

Forge tests passed on my mainnet fork.

### CODE REVIEW

**Comment by Drblessing** [comment]
Screenshot of it running in localhost in dev mode, had trouble building it with `npm run deploy` 

image

File: src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol [hunk 1]
```diff
@@ -13,10 +14,20 @@ contract TestUniswapLiquidity {
         uint _amountB
     ) external {
         IERC20(_tokenA).transferFrom(msg.sender, address(this), _amountA);
-        IERC20(_tokenB).transferFrom(msg.sender, address(this), _amountB);
+        // IERC201(_tokenB).transferFrom(msg.sender, address(this), _amountB);
```
Comment (t4sk) [inline src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol:17]: 1. Create a internal function called `safeTransferFrom`
2. Use `safeTransferFrom` for  both tokenA and B
3. Comment why `safeTransferFrom` is needed

File: src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol [hunk 2]
```diff
@@ -13,10 +14,20 @@ contract TestUniswapLiquidity {
...
+        // Manualy transfer from USDT
+        (bool successTransferFrom, ) = address(_tokenB).call(
+            abi.encodeCall(IERC20.transferFrom, (msg.sender, address(this), _amountB))
+        );
+        require(successTransferFrom, "USDT transferFrom failed");
 
         IERC20(_tokenA).approve(ROUTER, _amountA);
-        IERC20(_tokenB).approve(ROUTER, _amountB);
```
Comment (t4sk) [inline src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol:19]: Similar to comment above about `safeTransferFrom`

**Review by t4sk — CHANGES_REQUESTED** [review]

**Comment by t4sk** [comment]
Thanks for the PR. Please address the comments above

File: src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol (hunk 2 above)
Comment (Drblessing) [reply src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2Liquidity.sol:19]: Thanks, added

**Comment by Drblessing** [comment]
Thanks for the feedback, your comments have been addressed

File: src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2LiquidityTest.sol [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     * This function ensures that we are never blocked by a token's transferFrom implementation.
+     * Without it, the transferFrom call would fail.
+     */
+    function safeTransferFrom(
+        IERC20 token,
+        address sender,
+        address recipient,
+        uint amount
+    ) internal {
+        (bool successTransferFrom, ) = address(token).call(
```
Comment (t4sk) [inline src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2LiquidityTest.sol:61]: The point of `safeTransfer` is that the transfer function may or may not return a `bool`. So it needs to check if a data was returned or not.  Here is an example.

https://github.com/primitivefinance/rmm-core/blob/8d3ef9b4612af9212fff1da2fe15a99d59338c35/contracts/libraries/Transfers.sol#L12-L20

Please update accordingly

File: src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2LiquidityTest.sol [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        require(successTransferFrom, "transferFrom failed");
+    }
+
+    /**
+     * @dev Required for tokens that do not follow the ERC20 standard.
+     * For example, USDT approve doesn't return a bool.
+     * This function ensures that we are never blocked by a token's approve implementation.
+     * Without it, the approve call would fail.
+     */
+    function safeApprove(IERC20 token, address spender, uint amount) internal {
```
Comment (t4sk) [inline src/pages/defi/uniswap-v2-add-remove-liquidity/UniswapV2LiquidityTest.sol:73]: Similar comment as `safeTransfer`. Need to check the presence of return data

**Review by t4sk — CHANGES_REQUESTED** [review]

**Comment by Drblessing** [comment]
Added the new checks, thanks

**Pull request merged**